fus.c

```diff
@@ -670,10 +677,22 @@ mask_solvable_bare_rpms (Pool *pool)
 
           Id *mp = pool_whatprovides_ptr (pool, bare_rpms_rel);
           for (; *mp; mp++)
-            selection_make (pool,
-                            &selection,
-                            pool_solvid2str (pool, *mp),
-                            SELECTION_CANON | SELECTION_ADD);
+            {
+              Solvable *pkg = pool_id2solvable (pool, *mp);
+              /* The query above looks for all provides. But we only want to
+               * mask the package if the name is identical. */
+              if (modpkg->name == pkg->name)
+                {
+                  g_debug ("Excluding %s (masked by %s in %s)",
+                          pool_solvable2str (pool, pkg),
+                          pool_solvable2str (pool, modpkg),
+                          pool_solvable2str (pool, s));
+                  selection_make (pool,
+                          &selection,
+                          pool_solvid2str (pool, *mp),
+                          SELECTION_CANON | SELECTION_ADD);
```

Comment:
There's probably a better way to make this query. Essentially what's needed is to find solvables with name identical to `modpkg->name` that do not provide `modular-package()`.

Reply:
on RPM level you could do `(name WITHOUT modular-package())` but I won't find time to look into this code ever…